packages/edit-post/src/editor.js

```diff
@@ -24,6 +26,14 @@ import { unlock } from './lock-unlock';
 import useNavigateToEntityRecord from './hooks/use-navigate-to-entity-record';
 
 const { ExperimentalEditorProvider } = unlock( editorPrivateApis );
+const { BlockRemovalWarningModal } = unlock( blockEditorPrivateApis );
```

Comment:
Instead of duplicating this logic between `edit-post` and `edit-site`, why not add it to the `EditorProvider` instead?